### PR TITLE
Drop Python 3.6 support

### PR DESCRIPTION
**Pull request by rebeccafair**
Addresses #232 

Mainly simple changes, I have also added a py37-min test to the release tests, so we can make sure the PyPI wheel works against older versions of Numpy.

### CODE REVIEW

**Comment by rebeccafair** [comment]
Ok, so testing minimum dependencies (`numpy 1.14.5`, `scipy 1.1.0`, `matplotlib 2.2.0`m `h5py 2.8.0`, `pint 0.10.1`, all other dependencies unchanged) with Python 3.7 is giving me the following error, both on CI (Linux) and locally (Windows):
```
>               import euphonic._euphonic as euphonic_c
E               ImportError: numpy.core.multiarray failed to import
../../.tox/py37-minrequirements-linux/lib/python3.7/site-packages/euphonic/force_constants.py:631: ImportError
```
This error seems to persist (even on versions of Euphonic downloaded from PyPI) unless I use a version of `numpy >= 1.20.0`.

The strange thing is, if I try testing on the minimum versions with Python 3.8 (`numpy 1.17.3`, `scipy 1.3.2`, `matplotlib 2.2.5`, `pint 0.10.1`, `h5py 2.10.0`, versions slightly higher as I've used the ones which have wheels on PyPI, all other versions unchanged) it works fine! `numpy >= 1.20.0` seems to be required for Python 3.7, but not 3.8.

I can't find anything in the Numpy `1.19.5 -> 1.20.0` changelog which would suggest what has changed. Bewildering...

**Comment by ajjackson** [comment]
Odd stuff. So we _could_ avoid the problem by skipping straight to Py3.8, but that's not a very satisfying solution.

**Comment by rebeccafair** [comment]
Interestingly, `numpy 1.20.0` is the version that officially dropped Python 3.6 support and moved onto 3.7. Maybe there were some bugs there that weren't noticed until that version, hidden somewhere in https://github.com/numpy/numpy/compare/v1.19.5...v1.20.0.

I'll try and find a fix but if it takes me more than a few hours we might have to think of something else. Specify dependencies depending on the Python version? If Python 3.7 require `numpy >= 1.20.0` and if Python 3.8 require `numpy>=1.17.3`?

**Comment by rebeccafair** [comment]
Doing some digging, `1.20` does have C API changes https://github.com/numpy/numpy/blob/66d33faabf1f19d4b192c38e927dfa727b0ed61f/doc/source/release/1.20.0-notes.rst#c-api-changes, for which it says to compile against the lowest version of Numpy you want to support, which we are already doing, so this shouldn't be an issue! I've also found from this commit https://github.com/numpy/numpy/commit/abca5ca34595fda048e16760cb6355a67220f5db there are some specific workarounds for < Python 3.8, so maybe this explains why we're having issues with 3.7 and not 3.8, but really I have no idea...

**Comment by rebeccafair** [comment]
Oops, turns out the build directory from previous Python 3.7 tests wasn't being deleted for the minimum version tests, so the C extension wasn't actually being rebuilt with a lower version of Numpy... Why this was a problem with 3.7 and not 3.6 I have no idea, I'm not going to worry about it. Fix is in [e65706b](https://github.com/pace-neutrons/Euphonic/pull/237/commits/e65706bf6a57965e94498da32af90b58e3b9e9e9).

File: tests_and_analysis/test/euphonic_test/test_crystal.py [hunk 1]
```diff
@@ -449,3 +447,21 @@ def test_get_symmetry_equivalent_atoms(
...
             npt.assert_equal(equiv_atoms, exp_equiv_atoms)
+
+    @pytest.mark.parametrize('crystal, kwargs, expected_symprec', [
+        (get_crystal('LZO'), {'tol': 1e-12*ureg('angstrom')}, 1e-12),
+        (get_crystal('LZO'), {'tol': 1.8897261246e-12*ureg('bohr')}, 1e-12)
+        ])
+    def test_get_symmetry_equivalent_atoms_with_tol(
+            self, mocker, crystal, kwargs, expected_symprec):
+        # Ensure tol gets passed correctly to spglib - we don't care about
+        # the details of what spglib does
```
Comment (ajjackson) [inline tests_and_analysis/test/euphonic_test/test_crystal.py:458]: Hurrah, less data in the repo!

**Review by ajjackson — CHANGES_REQUESTED** [review]
Looks good!

I think we might as well remove that backport for now; if we need it later it will be with an API change and different version.

File: euphonic/util.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 import numpy as np
 import seekpath
 from seekpath.hpkot import SymmetryDetectionError
-from importlib_resources import open_text  # Backport for Python 3.6
+from importlib_resources import open_text
```
[the remark this answers is not in the document]
Comment (rebeccafair) [reply euphonic/util.py:14]: This is addressed in #238! Happy to merge this as-is for now?

**Review by ajjackson — APPROVED** [review]

**Pull request merged**